src/Components/test/E2ETest/ServerExecutionTests/RenderingModesTests.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.ServerExecutionTests
+{
+    public class RenderingModesTest : ServerTestBase<AspNetSiteServerFixture>
+    {
+        public RenderingModesTest(
+            BrowserFixture browserFixture,
+            AspNetSiteServerFixture serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+            _serverFixture.Environment = AspNetEnvironment.Development;
+            _serverFixture.BuildWebHostMethod = TestServer.Program.BuildWebHost;
+        }
+
+        [Fact]
+        public void PrerenderedCircuitsShareSameCircuit()
+        {
+            Navigate("/startmodes/startmodeshost/prerendered");
+            Browser.Equal("prerendered", () => Browser.FindElement(By.CssSelector("#prerendered-1>h3")).Text);
+            Browser.Equal("prerendered", () => Browser.FindElement(By.CssSelector("#prerendered-2>h3")).Text);
+
+            BeginInteractivity();
+            var buttons = Browser.FindElements(By.CssSelector(".reveal"));
+            foreach (var button in buttons)
+            {
+                button.Click();
+            }
+
+            var ids = Browser.FindElements(By.CssSelector(".circuit-id")).Select(e => e.Text).ToArray();
+
+            Assert.Equal(2, ids.Length);
+            Assert.Single(ids.Distinct());
```

Comment:
I'm not sure this test is doing anything to wait until the UI updates before reading the displayed circuit IDs.

Reply:
Same with the other two tests below.